zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/FileSnap.java

```diff
@@ -226,13 +236,42 @@ public synchronized void serialize(DataTree dt, Map<Long, Integer> sessions, Fil
                 FileHeader header = new FileHeader(SNAP_MAGIC, VERSION, dbId);
                 serialize(dt, sessions, oa, header);
                 SnapStream.sealStream(snapOS, oa);
```

Comment:
if digest enabled, it need check sum twice? Does it  serialize zxid digest firstly, then write checksum? like this:
                serialize(dt, sessions, oa, header);
                //SnapStream.sealStream(snapOS, oa);

                // Digest feature was added after the CRC to make it backward
                // compatible, the older code cal still read snapshots which
                // includes digest.
                //
                // To check the intact, after adding digest we added another
                // CRC check.
                if (dt.serializeZxidDigest(oa)) {
                    //SnapStream.sealStream(snapOS, oa);
                    LOG.warn(" serializeZxidDigest success");
                }
                SnapStream.sealStream(snapOS, oa); // do it finally